Modules/nismodule.c

```diff
@@ -444,26 +479,28 @@ PyDoc_STRVAR(nis__doc__,
 
 static struct PyModuleDef nismodule = {
     PyModuleDef_HEAD_INIT,
-    "nis",
-    nis__doc__,
-    -1,
-    nis_methods,
-    NULL,
-    NULL,
-    NULL,
-    NULL
+    .m_name = "nis",
+    .m_doc = nis__doc__,
+    .m_size = sizeof(nis_state),
+    .m_methods = nis_methods,
+    .m_traverse = nis_traverse,
+    .m_clear = nis_clear,
+    .m_free = nis_free,
 };
 
 PyMODINIT_FUNC
 PyInit_nis(void)
 {
     PyObject *m, *d;
     m = PyModule_Create(&nismodule);
-    if (m == NULL)
+    if (m == NULL) {
         return NULL;
+    }
     d = PyModule_GetDict(m);
-    NisError = PyErr_NewException("nis.error", NULL, NULL);
-    if (NisError != NULL)
-        PyDict_SetItemString(d, "error", NisError);
+    nis_state* state = get_nis_state(m);
+    state->nis_error = PyErr_NewException("nis.error", NULL, NULL);
+    if (state->nis_error != NULL) {
```

Comment:
This error checking code is weird. While you didn't write it, would you mind to take the opportunity to fix it? Something like:

```
if (state->nis_error == NULL) {
    goto error;
}

(...)

error:
    Py_DECREF(m);
    return NULL;
```